hashintel/hash
# H-759: Add bulk actions dropdown on entities and types tables

## 🌟 What is the purpose of this PR?




This PR adds a bulk actions dropdown on the various entities and types tables in the HASH app, which currently allows for bulk archiving/un-archiving entity/property/link types, and page entities.

## 🔗 Related links





- H-759

## 🔍 What does this change?




- updates the `queryPropertyTypes` GQL resolver to accept `includeArchive` and `latestOnly` arguments, to be consistent with the `queryEntityTypes` GQL resolver and enable the FE to retrieve archived property types

## Pre-Merge Checklist 🚀

### 🚢 Has this modified a publishable library?




This PR:


### 📜 Does this require a change to the docs?




The changes in this PR:


### 🕸️ Does this require a change to the Turbo Graph?




The changes in this PR:


## ⚠️ Known issues



## 🐾 Next steps



- [Allow bulk actions to be applied to a subset of the current selection, and update the tooltip to explain this](https://linear.app/hash/issue/H-824/allow-bulk-actions-to-be-applied-to-a-subset-of-the-current-selection)

## 🛡 What tests cover this?



None.

## ❓ How to test this?



1. Go to `/pages` page, and ensure pages can be bulk selected and archived/un-archived in the entities table
2. Go to the `/entities` page, and ensure entities can be bulk selected. Note that only pages can be archived, as general entity archival is not yet supported.
3. Go to the `/types` page and ensure entity types, link types and property types can be bulk selected and archived/un-archived. Note that data type archival is not yet supported, as we don't yet have custom data types.

## 📹 Demo




https://github.com/hashintel/hash/assets/42802102/ebde0289-403f-4373-94a1-b8c631cdd052
